tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,208 @@
+import math
+import pytest
+from decimal import Decimal
+import datetime
+from datetime import timezone, timedelta
+import trino
+
+
+@pytest.fixture
+def trino_connection(run_trino):
+    _, host, port = run_trino
+
+    yield trino.dbapi.Connection(
+        host=host, port=port, user="test", source="test", max_attempts=1
+    )
+
+
+def test_int_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="false", python=False) \
+        .add_field(sql="cast(null AS BOOLEAN)", python=None) \
+        .add_field(sql="cast(127 AS TINYINT)", python=127) \
+        .add_field(sql="cast(-128 AS TINYINT)", python=-128) \
+        .add_field(sql="cast(null AS TINYINT)", python=None) \
+        .add_field(sql="cast(32767 AS SMALLINT)", python=32767) \
+        .add_field(sql="cast(-32768 AS SMALLINT)", python=-32768) \
+        .add_field(sql="cast(null AS SMALLINT)", python=None) \
+        .add_field(sql="cast(2147483647 AS INTEGER)", python=2147483647) \
+        .add_field(sql="cast(-2147483648 AS INTEGER)", python=-2147483648) \
+        .add_field(sql="cast(null AS INTEGER)", python=None) \
+        .add_field(sql="cast(9223372036854775807 AS BIGINT)", python=9223372036854775807) \
+        .add_field(sql="cast(-9223372036854775808 AS BIGINT)", python=-9223372036854775808) \
+        .add_field(sql="cast(null AS BIGINT)", python=None) \
+        .execute()
+
+
+def test_float_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(3.4028235E38 AS REAL)", python=3.4028235e+38) \
+        .add_field(sql="cast(1.4E-45 AS REAL)", python=1.4e-45) \
+        .add_field(sql="cast('Infinity' AS REAL)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS REAL)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS REAL)", python=math.nan) \
+        .add_field(sql="cast(null AS REAL)", python=None) \
+        .add_field(sql="cast(1.7976931348623157E308 AS DOUBLE)", python=1.7976931348623157e+308) \
+        .add_field(sql="cast(4.9E-324 AS DOUBLE)", python=5e-324) \
+        .add_field(sql="cast('Infinity' AS DOUBLE)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS DOUBLE)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS DOUBLE)", python=math.nan) \
+        .add_field(sql="cast(null AS DOUBLE)", python=None) \
+        .add_field(sql="DECIMAL '10.3'", python=Decimal('10.3')) \
+        .add_field(sql="cast(null AS DECIMAL)", python=None) \
+        .add_field(sql="cast('0.123456789123456789' AS DECIMAL(18,18))", python=Decimal('0.123456789123456789')) \
+        .add_field(sql="cast(null AS DECIMAL(18,18))", python=None) \
+        .add_field(sql="cast('10.3' AS DECIMAL(38,0))", python=10.0) \
+        .add_field(sql="cast(null AS DECIMAL(38,0))", python=None) \
```

Comment:
Add tests for DECIMAL(38, 38) and something below DECIMAL(18, x) since above 18 everything is represented differently in Trino.

Reply:
DECIMAL(38,38) doesn't seem to be accepted (`DECIMAL scale must be in range [0, precision (18)]: 38`)

Reply:
It means we're dropping the precision. Since DECIMAL(38, x) means precision = 38 but the error message says precision is 18.

So a bug to be fixed.

Reply:
Note that this is a Trino error, not a Python error. `SELECT cast('10.3' AS DECIMAL(38,38))` blows up in Starburst.

Reply:
It's because the value is not 38,38. 38,38 means only 38 decimal places, so values < 0. Try to test with `CAST('0.3123' AS DECIMAL(38,38))`

Reply:
Done